### DISCUSSION TITLE
Add VAAPI HW transcoding to Scrypted LXC

### DISCUSSION THREAD

**Discussion by VCTGomes** [tteck/Proxmox#1603] [Ideas]
Scrypted has support to transcoding cameras with VAPPI and others protocols, but it doesn't work with this script, unfortunately.

It's important because sometimes we need to transcode some camera in another format, like from H265 to H264 to use on HomeKit.

Plex LXC can handle and add easily VAAPI when it's privileged, so I think it could be easily to replicate

**Comment by kzaoaai** [comment]
would definitely be nice to use the iGPU with transcoding and the OpenVino plugin.

**Comment by tteck** [comment]
This feature can be tested for the next 24 hours. Based on feedback, there is a possibility that it might be incorporated into the main branch. Additionally, it will enable USB device passthrough for Emby, Jellyfin, Plex, and Tdarr.

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/scrypted.sh)"
```

**Reply by VCTGomes** [reply]
Really happy to see a so super quickly reply. I tried to install it, but unfortunately I got the error when try to run the scrip. Container was created, but Scrypted isn't installed yet.

```
 [ERROR] in line 20: exit code 100: while executing command $STD apt-get install -y avahi-daemon
++ [[ 20 -eq 24 ]]
+ exit

```

**Reply by tteck** [reply]
Debian 11?

**Reply by tteck** [reply]
So, it appears to work with Debian 12 also.

**Comment by VCTGomes** [comment]
> This feature can be tested for the next 24 hours. Based on feedback, there is a possibility that it might be incorporated into the main branch. Additionally, it will enable USB device passthrough for Emby, Jellyfin, Plex, and Tdarr.
> 
> ```
> bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/dev/ct/scrypted.sh)"
> ```



> Debian 11?

I'm sorry, maybe I selected something wrong. Now it's installed, but vainfo command isn't working yet

<img width="885" alt="image" src="https://github.com/tteck/Proxmox/assets/2366600/38be562b-b199-48aa-b155-374a971e6181">

**Reply by tteck** [reply]
```
root@scrypted:~# vainfo
error: XDG_RUNTIME_DIR is invalid or not set in the environment.
error: can't connect to X server!
libva info: VA-API version 1.17.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_17
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.17 (libva 2.12.0)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 23.1.1 ()
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSliceLP
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSliceLP
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointEncPicture
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSliceLP
      VAProfileVP8Version0_3          : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointVLD
      VAProfileHEVCMain10             : VAEntrypointVLD
      VAProfileVP9Profile0            : VAEntrypointVLD
      VAProfileVP9Profile2            : VAEntrypointVLD
```

**Reply by VCTGomes** [reply]
Let me try again

**Reply by kzaoaai** [reply]
it worked for me. Even the Openvino plugin shows GPU as an option. I even kept the same container from before, but borrowed the "Setting Up Hardware Acceleration" section of the script to make it work. Obviously, the container has to be privileged with the mapping similar to the Plex lxc.

**Comment by tteck** [comment]
Incorporated into the main branch

**Comment by Michelinu** [comment]
VAAPI transcoding seems to be broken.
Once selected in H264 Encoder Arguments, it does revert back to libx64 (Software).

Here's the output of `vainfo`:
```
root@scrypted:~# vainfo
error: XDG_RUNTIME_DIR is invalid or not set in the environment.
error: can't connect to X server!
libva info: VA-API version 1.17.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_17
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.17 (libva 2.12.0)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 23.1.1 ()
vainfo: Supported profile and entrypoints
      VAProfileNone                   : VAEntrypointVideoProc
      VAProfileNone                   : VAEntrypointStats
      VAProfileMPEG2Simple            : VAEntrypointVLD
      VAProfileMPEG2Main              : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointVLD
      VAProfileH264Main               : VAEntrypointEncSliceLP
      VAProfileH264High               : VAEntrypointVLD
      VAProfileH264High               : VAEntrypointEncSliceLP
      VAProfileJPEGBaseline           : VAEntrypointVLD
      VAProfileJPEGBaseline           : VAEntrypointEncPicture
      VAProfileH264ConstrainedBaseline: VAEntrypointVLD
      VAProfileH264ConstrainedBaseline: VAEntrypointEncSliceLP
      VAProfileVP8Version0_3          : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointVLD
      VAProfileHEVCMain               : VAEntrypointEncSliceLP
      VAProfileHEVCMain10             : VAEntrypointVLD
      VAProfileHEVCMain10             : VAEntrypointEncSliceLP
      VAProfileVP9Profile0            : VAEntrypointVLD
      VAProfileVP9Profile0            : VAEntrypointEncSliceLP
      VAProfileVP9Profile1            : VAEntrypointVLD
      VAProfileVP9Profile1            : VAEntrypointEncSliceLP
      VAProfileVP9Profile2            : VAEntrypointVLD
      VAProfileVP9Profile2            : VAEntrypointEncSliceLP
      VAProfileVP9Profile3            : VAEntrypointVLD
      VAProfileVP9Profile3            : VAEntrypointEncSliceLP
      VAProfileHEVCMain12             : VAEntrypointVLD
      VAProfileHEVCMain422_10         : VAEntrypointVLD
      VAProfileHEVCMain422_12         : VAEntrypointVLD
      VAProfileHEVCMain444            : VAEntrypointVLD
      VAProfileHEVCMain444            : VAEntrypointEncSliceLP
      VAProfileHEVCMain444_10         : VAEntrypointVLD
      VAProfileHEVCMain444_10         : VAEntrypointEncSliceLP
      VAProfileHEVCMain444_12         : VAEntrypointVLD
      VAProfileHEVCSccMain            : VAEntrypointVLD
      VAProfileHEVCSccMain            : VAEntrypointEncSliceLP
      VAProfileHEVCSccMain10          : VAEntrypointVLD
      VAProfileHEVCSccMain10          : VAEntrypointEncSliceLP
      VAProfileHEVCSccMain444         : VAEntrypointVLD
      VAProfileHEVCSccMain444         : VAEntrypointEncSliceLP
      VAProfileAV1Profile0            : VAEntrypointVLD
      VAProfileHEVCSccMain444_10      : VAEntrypointVLD
      VAProfileHEVCSccMain444_10      : VAEntrypointEncSliceLP
```

OpenVINO plugin does recognize it.
